### PR TITLE
[example] [refactor] Upgrade to ti.field syntax for many examples

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1500

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: examples/cg_possion.py [hunk 1]
```diff
@@ -6,10 +6,10 @@
 steps = 32
 eps = 1e-5
 
-b = ti.field(ti.f32, (n, n))
-x = ti.field(ti.f32, (n, n))
-d = ti.field(ti.f32, (n, n))
-r = ti.field(ti.f32, (n, n))
+b = ti.field(float, (n, n))
```
Comment (Rullec) [inline examples/cg_possion.py:9]: simply have a tiny question ;-).

Has the type system `ti.xxx` been not recommened anymore in Taichi-scope? I checked the [doc](https://taichi.readthedocs.io/en/stable/type.html), it seems here is not evidence.

**Comment by Rullec** [comment]
Nice job! 
Just now I search for `dt=` in the `examples/` part again, and it shows maybe we still need to do some replacement further more.
```
./autodiff_minimization.py:5:x = ti.var(dt=ti.f32, shape=n, needs_grad=True)
./autodiff_minimization.py:6:y = ti.var(dt=ti.f32, shape=n)
./autodiff_minimization.py:7:L = ti.var(dt=ti.f32, shape=(), needs_grad=True)
./cornell_box.py:9:color_buffer = ti.Vector(3, dt=ti.f32, shape=res)
./euler.py:23:Q = ti.Vector(4, dt=real,
./euler.py:25:Q_old = ti.Vector(4, dt=real, shape=(N, N))
./euler.py:26:W = ti.Vector(4, dt=real, shape=(N, N))  # [rho, u, v, p] cell avg
./euler.py:27:W_xl = ti.Vector(4, dt=real, shape=(N, N, 3))  # left side of x-face
./euler.py:28:W_xr = ti.Vector(4, dt=real, shape=(N, N, 3))  # right side of x-face
./euler.py:29:W_yl = ti.Vector(4, dt=real, shape=(N, N, 3))  # left side of y-face
./euler.py:30:W_yr = ti.Vector(4, dt=real, shape=(N, N, 3))  # right side of y-face
./euler.py:31:F_x = ti.Vector(4, dt=real, shape=(N, N))  # x-face flux
./euler.py:32:F_y = ti.Vector(4, dt=real, shape=(N, N))  # y-face flux
./euler.py:33:dt = ti.var(dt=real, shape=())
./euler.py:34:img = ti.var(dt=ti.f32, shape=(res, res))
./export_ply.py:10:rgba = ti.Vector(4, dt=ti.f32, shape=(10, 10, 10))
./export_ply.py:9:pos = ti.Vector(3, dt=ti.f32, shape=(10, 10, 10))
./mgpcg.py:21:r = [ti.var(dt=real) for _ in range(n_mg_levels)]  # residual
./mgpcg.py:22:z = [ti.var(dt=real) for _ in range(n_mg_levels)]  # M^-1 r
./mgpcg.py:23:x = ti.var(dt=real)  # solution
./mgpcg.py:24:p = ti.var(dt=real)  # conjugate gradient
./mgpcg.py:25:Ap = ti.var(dt=real)  # matrix-vector product
./mgpcg.py:26:alpha = ti.var(dt=real)  # step size
./mgpcg.py:27:beta = ti.var(dt=real)  # step size
./mgpcg.py:28:sum = ti.var(dt=real)  # storage for reductions
./mgpcg.py:29:pixels = ti.var(dt=real, shape=(N_gui, N_gui))  # image buffer
./mgpcg_advanced.py:27:        self.r = [ti.var(dt=real) for _ in range(self.n_mg_levels)]  # residual
./mgpcg_advanced.py:28:        self.z = [ti.var(dt=real)
./mgpcg_advanced.py:30:        self.x = ti.var(dt=real)  # solution
./mgpcg_advanced.py:31:        self.p = ti.var(dt=real)  # conjugate gradient
./mgpcg_advanced.py:32:        self.Ap = ti.var(dt=real)  # matrix-vector product
./mgpcg_advanced.py:33:        self.alpha = ti.var(dt=real)  # step size
./mgpcg_advanced.py:34:        self.beta = ti.var(dt=real)  # step size
./mgpcg_advanced.py:35:        self.sum = ti.var(dt=real)  # storage for reductions
./mgpcg_advanced.py:36:        self.pixels = ti.var(dt=real,
./odop_solar.py:11:        self.x = ti.Vector(2, dt=ti.f32, shape=n)
./odop_solar.py:12:        self.v = ti.Vector(2, dt=ti.f32, shape=n)
./odop_solar.py:13:        self.center = ti.Vector(2, dt=ti.f32, shape=())
./particle_renderer.py:12:color_buffer = ti.Vector(3, dt=ti.f32)
./particle_renderer.py:13:bbox = ti.Vector(3, dt=ti.f32, shape=2)
./particle_renderer.py:14:grid_density = ti.var(dt=ti.i32)
./particle_renderer.py:15:voxel_has_particle = ti.var(dt=ti.i32)
./particle_renderer.py:19:particle_x = ti.Vector(3, dt=ti.f32)
./particle_renderer.py:20:particle_v = ti.Vector(3, dt=ti.f32)
./particle_renderer.py:21:particle_color = ti.Vector(3, dt=ti.f32)
./pbf2d.py:52:old_positions = ti.Vector(dim, dt=ti.f32)
./pbf2d.py:53:positions = ti.Vector(dim, dt=ti.f32)
./pbf2d.py:54:velocities = ti.Vector(dim, dt=ti.f32)
./pbf2d.py:61:position_deltas = ti.Vector(dim, dt=ti.f32)
./pbf2d.py:63:board_states = ti.Vector(2, dt=ti.f32)
./quadtree.py:19:img = ti.Vector(3, dt=ti.f32, shape=(RES, RES))
./sdf2d.py:7:img = ti.var(dt=ti.f32, shape=(N, N))
./sdf2d.py:8:light_pos = ti.Vector(2, dt=ti.f32, shape=())
./sdf_renderer.py:8:color_buffer = ti.Vector(3, dt=ti.f32, shape=res)
./simple_uv.py:7:pixels = ti.Vector(3, dt=ti.f32, shape=res)
./waterwave.py:14:pixels = ti.var(dt=ti.f32, shape=shape)
./waterwave.py:15:background = ti.var(dt=ti.f32, shape=shape)
./waterwave.py:16:height = ti.var(dt=ti.f32, shape=shape)
./waterwave.py:17:velocity = ti.var(dt=ti.f32, shape=shape)
./waterwave.py:18:acceleration = ti.var(dt=ti.f32, shape=shape)
```

File: examples/cg_possion.py (hunk 1 above)
Comment (archibate) [reply examples/cg_possion.py:9]: Sorry about the doc, please checkout the [Default Precisions](https://taichi.readthedocs.io/en/stable/type.html#default-precisions) chapter:
```py
ti.init(default_ip=ti.i64, default_fp=ti.f32)  # set default precisions on ti.init()

x = ti.var(float, 5)
y = ti.var(int, 5)
# is equivalent to:
x = ti.var(ti.f32, 5)
y = ti.var(ti.i64, 5)

def func(a: float) -> int:  # use float and int as aliases for default precision
    ...

# is equivalent to:
def func(a: ti.f32) -> ti.i64:
    ...
```
For example, users can run `TI_DEFAULT_FP=64 python examples/fractal.py` for running fractal in float64 precision without changing it's source code.

Comment (Rullec) [reply examples/cg_possion.py:9]: Thanks! I get it. It's a lot of convenience, that we don't need to specify `ti.xxx` everytime. We can recommend our users  do coding in this way as well.

Comment (archibate) [reply examples/cg_possion.py:9]: Yes, feel free to advocate it more in doc if you'd like to.

**Comment by archibate** [comment]
> Just now I search for dt= in the examples/ part again, and it shows maybe we still need to do some replacement further more.

Yes, I'll do iapr. If you think this PR is OK for now, let's merge it, wdyt?

**Pull request merged**